modules/Bitfinex.py

```diff
@@ -0,0 +1,322 @@
+        '''
+        The ticker is a high level overview of the state of the market
+        https://bitfinex.readme.io/v1/reference#rest-public-ticker
+        '''
+        t = int(time.time())
+        if (t - self.tickerTime < 60):
+            return self.ticker
+
+        for symbol in self._getSymbols():
+            base = symbol[3:].upper()
+            curr = symbol[:3].upper()
+            if base in ['BTC', 'USD'] and (curr == 'BTC' or curr in self.usedCurrencies):
+                couple = (base + '_' + curr)
+                coupleReverse = (curr + '_' + base)
+
+                try:
+                    ticker = self._get('pubticker/'+symbol)
+                except ApiError as e:
+                    if not self.ticker[couple]:
+                        raise e
+
+                self.ticker[couple] = {
+                    "last": ticker['last_price'],
+                    "lowestAsk": ticker['ask'],
+                    "highestBid": ticker['bid'],
+                    "percentChange": "",
+                    "baseVolume": str(float(ticker['volume'])*float(ticker['mid'])),
+                    "quoteVolume": ticker['volume']
+                }
+                self.ticker[coupleReverse] = {
+                    "last": 1 / float(self.ticker[couple]['last']),
+                    "lowestAsk": 1 / float(self.ticker[couple]['lowestAsk']),
+                    "highestBid": 1 / float(self.ticker[couple]['highestBid'])
+                }
+
+        self.tickerTime = t
+
+        return self.ticker
+
+    def return_available_account_balances(self, account):
+        '''
+        Returns own balances sorted by account
+        https://bitfinex.readme.io/v1/reference#rest-auth-wallet-balances
+        '''
+        signed_payload = self._sign_payload({
+            "request": "/v1/balances",
+            "nonce": self._nonce
+        })
+
+        bfxResp = self._post("balances", signed_payload)
+        balances = Bitfinex2Poloniex.convertAccountBalances(bfxResp, account)
+
+        if balances['lending']:
+            self.usedCurrencies = []
+            for curr in balances['lending']:
+                self.usedCurrencies.append(curr)
+
+        return balances
+
+    def cancel_loan_offer(self, currency, order_number):
+        '''
+        Cancels an offer
+        https://bitfinex.readme.io/v1/reference#rest-auth-cancel-offer
+        '''
+        signed_payload = self._sign_payload({
+            "request": "/v1/offer/cancel",
+            "offer_id": order_number,
+            "nonce": self._nonce
+        })
+
+        bfxResp = self._post("offer/cancel", signed_payload)
+
+        success = 0
+        message = ''
+        try:
+            if bfxResp['id'] == order_number:
+                success = 1
+                message = "Loan offer canceled ({:.4f} @ {:.4f}%).".format(float(bfxResp['remaining_amount']),
+                                                                           float(bfxResp['rate'])/365)
```

Comment:
By no means a requirement with this, but it's recommended in most circles and I think it would make other people do it in future.